app/vminsert/netstorage/insert_ctx.go

```diff
@@ -21,8 +22,9 @@ type InsertCtx struct {
 	Labels        []prompb.Label
 	MetricNameBuf []byte
 
-	bufRowss  []bufRows
-	labelsBuf []byte
+	groupBufRowss           map[string][]bufRows
```

Comment:
It is better from performance PoV to use a slice instead a map here, since access by numeric replication group index is faster than access by replication group name.